core/common/src/main/java/alluxio/resource/DynamicResourcePool.java

```diff
@@ -386,14 +373,7 @@ public int size() {
    * @return true if the pool is full
    */
   private boolean isFull() {
-    boolean full = false;
-    try {
-      mLock.lock();
-      full = mResources.size() >= mMaxCapacity;
-    } finally {
-      mLock.unlock();
-    }
-    return full;
+    return mResources.size() >= mMaxCapacity;
```

Comment:
I've removed that lock in size(). It is not necessary to guard size by lock here. Thanks